pkg/api/dataproxy.go

```diff
@@ -115,12 +118,28 @@ func ProxyDataSourceRequest(c *middleware.Context) {
 		}
 	}
 
+	outputToAuditLog(ds.Type, c)
+
 	proxy := NewReverseProxy(ds, proxyPath, targetUrl)
 	proxy.Transport, err = ds.GetHttpTransport()
 	if err != nil {
 		c.JsonApiErr(400, "Unable to load TLS certificate", err)
 		return
 	}
+
 	proxy.ServeHTTP(c.Resp, c.Req.Request)
 	c.Resp.Header().Del("Set-Cookie")
 }
+
+func outputToAuditLog(dataSourceType string, c *middleware.Context) {
+	auditLogger := log.New("data-proxy-audit", "userid", c.UserId, "orgid", c.OrgId, "username", c.Login)
+
+	var body string
+	if c.Req.Request.Body != nil {
+		buffer, _ := ioutil.ReadAll(c.Req.Request.Body)
+		c.Req.Request.Body = ioutil.NopCloser(bytes.NewBuffer(buffer))
+		body = string(buffer)
+	}
+
+	auditLogger.Info("Proxying incoming request", "datasource", dataSourceType, "uri", c.Req.RequestURI, "method", c.Req.Request.Method, "body", body)
```

Comment:
Using `Info` this will be enabled by default for everyone. I think that's to much noisy for most grafana users. But lowering it to `Debug` doesnt make sense either. I think there should be a setting in the grafana config to enable audit logging.